### PR TITLE
propagate transitions to d3fc-data-join

### PR DESCRIPTION
**Pull request by ColinEberhardt**

### CODE REVIEW

File: examples/chart.js [hunk 1]
```diff
@@ -33,6 +33,4 @@ function render() {
     if (typeof renderComponent === 'function') {
         renderComponent();
     }
-
-    requestAnimationFrame(render);
```
Comment (chrisprice) [inline examples/chart.js]: Does this not break the crosshair example?

File: package.json [hunk 2]
```diff
@@ -19,17 +19,17 @@
     "semantic-release": "semantic-release pre && npm publish && semantic-release post"
   },
   "devDependencies": {
-    "d3-axis": "^1.0.3",
-    "d3-shape": "^1.0.3",
+    "babel-polyfill": "^6.16.0",
```
Comment (chrisprice) [inline package.json]: Would it make sense to put this in d3fc-scripts? (Maybe just open an issue for now)

File: src/svg/gridline.js [hunk 3]
```diff
@@ -19,6 +19,11 @@ export default () => {
 
     const instance = (selection) => {
 
+        if (selection.selection) {
+            xJoin.transition(selection);
+            yJoin.transition(selection);
```
Comment (chrisprice) [inline src/svg/gridline.js]: Did you decide against transitioning the containers?

**Comment by ColinEberhardt** [comment]
@chrisprice I've updated the code, addressing those points.

The gridlines component really needs some logic similar to axis in order to make transitions work well. I think I'll raise an issue for that and address after d3fc v.next release. I'm happy that d3fc-series now supports transitions, which is the main one to solve.

**Comment by chrisprice** [comment]
Agreed, I think they probably all need it.

**Pull request merged**